libraries/chain/db_block.cpp

```diff
@@ -657,9 +657,10 @@ processed_transaction database::_apply_transaction(const signed_transaction& trx
       FC_ASSERT( trx.expiration <= now + chain_parameters.maximum_time_until_expiration, "",
                  ("trx.expiration",trx.expiration)("now",now)("max_til_exp",chain_parameters.maximum_time_until_expiration));
       FC_ASSERT( now <= trx.expiration, "", ("now",now)("trx.exp",trx.expiration) );
-      FC_ASSERT( head_block_time() <= HARDFORK_CORE_1573_TIME
-            || trx.get_packed_size() <= chain_parameters.maximum_transaction_size,
-            "Transaction exceeds maximum transaction size." );
+      if ( !(skip & skip_block_size_check ) ) // don't waste time on replay
```

Comment:
IMHO this code is wrong. `skip_block_size_check` flag is set when pushing pending transactions, that means the large transactions will propagate and will accumulate in RAM.

Reply:
Ooff! I will test and attempt a different solution. Thanks for the catch.

Reply:
OK, I was wrong, was misled by the comment here: https://github.com/bitshares/bitshares-core/blob/fa38ee46524f638d6cefde20d39e575a0e762000/libraries/chain/db_block.cpp#L215-L224

Actually we didn't set the flag when pushing transactions to pending queue:
https://github.com/bitshares/bitshares-core/blob/fa38ee46524f638d6cefde20d39e575a0e762000/libraries/app/application.cpp#L598

`push_transaction` is called with default `skip` parameter, aka `skip_nothing`.